docs/content/docs/guide/incoming_webhook.md

```diff
@@ -59,6 +59,156 @@ spec:
+kind: Repository
+metadata:
+  name: repo
+  namespace: ns
+spec:
+  url: "https://github.com/owner/repo"
+  incoming:
+    - targets:
+        - ^feature/.*  # Matches any branch starting with "feature/"
+      secret:
+        name: feature-webhook-secret
+      type: webhook-url
+```
+
+**Multiple webhooks with first-match-wins:**
+
+```yaml
+apiVersion: "pipelinesascode.tekton.dev/v1alpha1"
+kind: Repository
+metadata:
+  name: repo
+  namespace: ns
+spec:
+  url: "https://github.com/owner/repo"
+  incoming:
+    # Production - checked first (most specific)
+    - targets:
+        - main
+        - ^v[0-9]+\.[0-9]+\.[0-9]+$  # Semver tags like v1.2.3
+      secret:
+        name: prod-webhook-secret
+      params:
+        - prod_env
+      type: webhook-url
+    
+    # Feature branches - checked second
+    - targets:
+        - ^feature/.*
+      secret:
+        name: feature-webhook-secret
+      params:
+        - dev_env
+      type: webhook-url
+    
+    # Catch-all - checked last
+    - targets:
+        - ^.*$  # Matches any branch not caught above
+      secret:
+        name: default-webhook-secret
+      type: webhook-url
+```
+
+**Mix exact matches and regex patterns:**
+
+```yaml
+incoming:
+  - targets:
+      - main              # Exact match
+      - staging           # Exact match
+      - ^release/.*       # Regex pattern
+      - ^hotfix/.*        # Regex pattern
+    secret:
+      name: repo-incoming-secret
+    type: webhook-url
+```
+
+**Best Practices:**
+
+- Use anchors (`^` and `$`) in regex patterns to avoid unintended matches
+- Place production/sensitive webhooks first in the list
+- Use exact matches for known branches when possible (faster than regex)
+- Test regex patterns to ensure they match only intended branches
+
+#### Security Validation and Warnings
+
+Pipelines-as-Code automatically validates incoming webhook target patterns and logs security warnings to help you identify potentially dangerous configurations.
+
+**Automatic Security Checks:**
+
+When an incoming webhook request is processed, the system validates all target patterns and logs warnings for:
+
+1. **Catch-all patterns** that match all branches:
+   - `.*`, `.+`, `^.*$`, `^.+$`
+   - Warning: "pattern matches all branches - consider being more specific"
+
+2. **Unanchored sensitive patterns** that could expose production secrets:
+   - Patterns containing: `production`, `main`, `master`, `release`, `hotfix`, `staging`
+   - Without start anchor (`^`): May match branches like `reproduce-production-bug`
+   - Without end anchor (`$`): May match unintended branch suffixes
+
+3. **General unanchored patterns** that may match unintended branches
+
+**Example Security Warnings in Logs:**
+
+```text
+[WARN] [SECURITY] Repository 'my-repo' incoming webhook #1 target pattern 'production.*': 
+  pattern contains 'production' without start anchor (^) - may match unintended branches like 'reproduce-production'
```

Comment:
it it what hapening?  if the user had just - "dev" regexp.Match would now match thing like foo-dev-bar or other? 

That may going to break backward compatibility if that's the case